chain/network/src/peer/peer_actor.rs

```diff
@@ -151,37 +157,92 @@ impl PeerActor {
+        } else {
+            PeerMessage::deserialize(crate::network_protocol::Encoding::Borsh, msg)
+        }
+    }
+
+    fn send_message_or_log(&mut self, msg: &PeerMessage) {
+        if let Err(err) = self.send_message(msg) {
+            debug!(target: "network", "send_message(): {}", err);
+        }
+    }
+
+    fn send_message(&mut self, msg: &PeerMessage) -> anyhow::Result<()> {
+        if let Some(enc) = self.force_encoding {
+            return self.send_message_with_encoding(msg, enc);
+        }
+        if self.peer_status == PeerStatus::Connecting {
+            self.send_message_with_encoding(msg, crate::network_protocol::Encoding::Proto)?;
+            self.send_message_with_encoding(msg, crate::network_protocol::Encoding::Borsh)?;
+        } else if self.protocol_buffers_supported {
+            self.send_message_with_encoding(msg, crate::network_protocol::Encoding::Proto)?;
+        } else {
+            self.send_message_with_encoding(msg, crate::network_protocol::Encoding::Borsh)?;
+        }
+        Ok(())
+    }
+
+    fn send_message_with_encoding(
+        &mut self,
+        msg: &PeerMessage,
+        mode: crate::network_protocol::Encoding,
+    ) -> anyhow::Result<()> {
         // Skip sending block and headers if we received it or header from this peer.
         // Record block requests in tracker.
         match msg {
-            PeerMessage::Block(b) if self.tracker.has_received(b.hash()) => return,
+            PeerMessage::Block(b) if self.tracker.has_received(b.hash()) => return Ok(()),
             PeerMessage::BlockRequest(h) => self.tracker.push_request(*h),
             _ => (),
         };
 
-        match msg.try_to_vec() {
-            Ok(bytes) => {
-                self.tracker.increment_sent(bytes.len() as u64);
-                let bytes_len = bytes.len();
-                if !self.framed.write(bytes) {
-                    #[cfg(feature = "performance_stats")]
-                    let tid = near_rust_allocator_proxy::get_tid();
-                    #[cfg(not(feature = "performance_stats"))]
-                    let tid = 0;
-                    error!(
-                        "{} Failed to send message {} of size {}",
-                        tid,
-                        strum::AsStaticRef::as_static(msg),
-                        bytes_len,
-                    )
-                }
-            }
-            Err(err) => error!(target: "network", "Error converting message to bytes: {}", err),
-        };
+        let bytes = msg.serialize(mode)?;
+        // error!(target: "network", "Error converting message to bytes: {}", err),
+        self.tracker.increment_sent(bytes.len() as u64);
+        let bytes_len = bytes.len();
+        if !self.framed.write(bytes) {
+            #[cfg(feature = "performance_stats")]
+            let tid = near_rust_allocator_proxy::get_tid();
+            #[cfg(not(feature = "performance_stats"))]
+            let tid = 0;
+            bail!(
```

Comment:
Super minor, but `anyhow::bail!` reads a bit to me

Reply:
done